src/gsy_e/models/area/__init__.py

```diff
@@ -646,3 +672,17 @@ def is_market_balancing(self, market_id: str) -> bool:
     def is_market_future(self, market_id: str) -> bool:
         """Return True if market_id belongs to a FUTURE market."""
         return market_id == self.future_markets.id
+
+
+class Market(Area):
+    """Class to define geographical market areas that can contain children (areas or assets)."""
+
+
+class Asset(Area):
+    """Class to define assets (devices). These instances cannot contain children."""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        if self.children:
+            raise ValueError(f"{self.__class__.__name__} instances can't have children.")
```

Comment:
Just for future reviewers' clarification, I assume that `self.__class__.__name__` is here in order to support any future renames / rebranding of the class, right ?